### ISSUE TITLE
Infinite recursion slicing infinite Range

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#17859]
```julia
In [1]: r = Range(-oo, oo)                                                                                                                                    

In [2]: r[::2]                                                                                                                                                
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
<ipython-input-2-1d69a7d36af2> in <module>
----> 1 r[::2]

~/current/sympy/sympy/sympy/sets/fancysets.py in __getitem__(self, i)
    759                         stop if stop is None else -stop + 1:
    760                         start if start is None else -start:
--> 761                         step].reversed
    762                 #---------------------
    763                 # handle infinite on the left

~/current/sympy/sympy/sympy/sets/fancysets.py in __getitem__(self, i)
    759                         stop if stop is None else -stop + 1:
    760                         start if start is None else -start:
--> 761                         step].reversed
    762                 #---------------------
    763                 # handle infinite on the left

... last 2 frames repeated, from the frame below ...

~/current/sympy/sympy/sympy/sets/fancysets.py in __getitem__(self, i)
    759                         stop if stop is None else -stop + 1:
    760                         start if start is None else -start:
--> 761                         step].reversed
    762                 #---------------------
    763                 # handle infinite on the left

RecursionError: maximum recursion depth exceeded while calling a Python object
```
This should raise an error saying something like "slicing not possible on Range with infinite start" rather than `RecursionError`.

**Comment by arun-y99** [comment]
The `__get_item__(self,i)` method handles finite ranges as well as ranges with either infinite on the right or left. Does it handle infinity when it occurs on both left and right sides (as in the case of `Range(-oo,oo)`)?

**Comment by gschintgen** [comment]
Fixed by #17859 (which has "addresses issue #..." instead of "Fixes #...", so it didn't close automatically).
